blob/src/main/java/io/crate/http/netty/HttpBlobHandler.java

```diff
@@ -298,50 +312,57 @@ private void partialContentResponse(String range, HttpRequest request, String in
             }
-        } catch (NumberFormatException ex) {
-            logger.error("Couldn't parse Range Header", ex);
-            start = 0;
-            end = raf.length();
-        }
 
-        HttpResponse response = new DefaultHttpResponse(HTTP_1_1, PARTIAL_CONTENT);
-        HttpHeaders.setContentLength(response, end - start + 1);
-        response.headers().set(CONTENT_RANGE, "bytes " + start + "-" + end + "/" + raf.length());
-        setDefaultGetHeaders(response);
+            HttpResponse response = new DefaultHttpResponse(HTTP_1_1, PARTIAL_CONTENT);
+            HttpHeaders.setContentLength(response, end - start + 1);
+            response.headers().set(CONTENT_RANGE, "bytes " + start + "-" + end + "/" + raf.length());
+            setDefaultGetHeaders(response);
 
-        ctx.getChannel().write(response);
-        ChannelFuture writeFuture = transferFile(digest, raf, start, end - start + 1);
-        if (!HttpHeaders.isKeepAlive(request)) {
-            writeFuture.addListener(ChannelFutureListener.CLOSE);
+            ctx.getChannel().write(response);
+            ChannelFuture writeFuture = transferFile(digest, raf, start, end - start + 1);
+            if (!HttpHeaders.isKeepAlive(request)) {
+                writeFuture.addListener(ChannelFutureListener.CLOSE);
+            }
+        } catch (IOException e) {
+            logger.trace("Cannot get a random access file stream to read from {}", e);
+            throw e;
         }
     }
 
     private void fullContentResponse(HttpRequest request, String index, final String digest) throws  IOException {
         BlobShard blobShard = localBlobShard(index, digest);
-        final RandomAccessFile raf = blobShard.blobContainer().getRandomAccessFile(digest);
-        HttpResponse response = new DefaultHttpResponse(HTTP_1_1, OK);
-        HttpHeaders.setContentLength(response, raf.length());
-        setDefaultGetHeaders(response);
-        logger.trace("HttpResponse: {}", response);
-        ctx.getChannel().write(response);
-        ChannelFuture writeFuture = transferFile(digest, raf, 0, raf.length());
-        if (!HttpHeaders.isKeepAlive(request)) {
-            writeFuture.addListener(ChannelFutureListener.CLOSE);
+        try (final RandomAccessFile raf = blobShard.blobContainer().getRandomAccessFile(digest)) {
+            HttpResponse response = new DefaultHttpResponse(HTTP_1_1, OK);
+            HttpHeaders.setContentLength(response, raf.length());
+            setDefaultGetHeaders(response);
+            logger.trace("HttpResponse: {}", response);
+            ctx.getChannel().write(response);
+            ChannelFuture writeFuture = transferFile(digest, raf, 0, raf.length());
+            if (!HttpHeaders.isKeepAlive(request)) {
+                writeFuture.addListener(ChannelFutureListener.CLOSE);
+            }
+        } catch (IOException e) {
+            logger.trace("Cannot get a random access file stream to read from {}", e);
```

Comment:
I don't think the trace logging is useful here, all Exceptions are handled in the exceptionCaught method and already logged